addons/stock/tests/test_stock_flow.py

```diff
@@ -2542,6 +2542,41 @@ def test_validate_backorder(self):
         backorder.button_validate()
         self.assertEqual(backorder.state, 'done')
 
+    def test_picking_mixed_tracking_with_backorder(self):
+        self.productB.tracking = 'lot'
+        picking = self.env['stock.picking'].create({
+            'location_id': self.supplier_location,
+            'location_dest_id': self.stock_location,
+            'picking_type_id': self.picking_type_in,
+            'company_id': self.env.company.id,
+        })
+        self.env['stock.move'].create({
```

Comment:
```suggestion
        no_tracking_move = self.env['stock.move'].create({
```
Just to make things clearer and be sure we're updating the right move.